src/actions/tests/tabs.spec.js

```diff
@@ -12,6 +12,22 @@ const { getSelectedSource, getSourceTabs } = selectors;
 
 import { sourceThreadClient as threadClient } from "./helpers/threadClient.js";
 
+import { addTab, moveTab } from "../sources/tabs";
+
+test("addTab returns expected action", () => {
+  const source = "testSource";
+  const tabIndex = "testTabIndex";
+  const output = addTab(source, tabIndex);
+  expect(output).toEqual({ type: "ADD_TAB", source, tabIndex });
```

Comment:
i'm sure if this check gives us any gains, as we need no certainty the tab was actually added.

We should probably instead use a selector to check if the tab was actually added to the redux store.

Reply:
I'd say that having the tab added is an integration test of action and reducer. We still have to know where the error occured and that information seems easier to obtain from  two unit tests where we know where the fault is instead of having an integration tests where we would still have to hunt down the bug.

Reply:
The `addTab` action just returns an object, i'm not sure even if an error occurs from dispatching the action, you get an error back here.